### ISSUE TITLE
Keypair and wallet generation (backend only)

### ISSUE THREAD

**Issue by aesedepece** [witnet/sheikah#19]

**Comment by aesedepece** [comment]
This should implement what's described in this WIP candidate:
https://github.com/aesedepece/WIPs/blob/wip-adansdpc-hdwallets/wip-adansdpc-hdwallets.md

**Comment by aesedepece** [comment]
Let's get things clearer. This is a roughly functional approach to keys and addresses derivation:

```typescript
const mnemonics: string = "..."
const seed: Seed = deriveSeedFromMnemonics(mnemonics)

// This is common to all blockchains
const masterKey: Buffer = privateKeyFactory(seed.masterSecret as Buffer, seed.chainCode as number)

// This is Bitcoin specific (BIP-0032) but it ignores BIP-0044
const wallet: Buffer = privateKeyFactory(masterKey, 0)
const externalChain: Buffer = privateKeyFactory(wallet, 0)
const addressPrivateKey: Buffer = privateKeyFactory(externalChain, 0)
const addressPublicKey: Buffer = derivePublicFromPrivate(addressPrivateKey)
const address: string = deriveAddressFromPublic(addressPublicKey)

// This is Witnet specific (WIP-0003)
const witnetAddressPrivateKey: Buffer = privateKeyFactory(privateKeyFactory(privateKeyFactory(privateKeyFactory(privateKeyFactory(masterKey, 0x8000000003), 0x8000001337), 0x80000000), 0), 0)
const witnetAddressPublicKey: Buffer = derivePublicFromPrivate(witnetAddressPrivateKey)
const witnetAddress: string = deriveAddressFromPublic(witnetAddressPublicKey)

//This is nicer
const witnetAddressPrivateKey: Buffer = deepPrivateKeyFactory(masterKey, "m/3'/4919'/0'/0/0")
```
Warning: this is typed pseudocode, not actual typescript!

**Comment by aesedepece** [comment]
`Buffer`s representing keys may be wrapped in a type alias for convenience.

**Comment by kronolynx** [comment]
closed by #66
